### PR TITLE
fix(tests) Add tox envs for Django 1.10,1.11,2.0

### PR DESCRIPTION
**Pull request by clemfromspace**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no 
| BC breaks?        | no     
| Related Issue     | None
| Need Doc update   | no

## What was changed

- [x] Updated the `tox.ini` and `travis.yml` files, adding the env for the `1.10`, `1.11` and `2.0` Django versions.
- [x] Fixed two tests that was failing on Django `2.0` (not impacting the previous versions).

## Why it was changed

So we are sure we are compatibles with the version stated in the README :)

### CODE REVIEW

File: tests/test_commands.py [hunk 1]
```diff
@@ -89,8 +89,11 @@ def test_clearindex(self):
             self.assertRegexpMatches(result, regex)
 
     def test_clearindex_with_args(self):
-        call_command('algolia_clearindex', stdout=self.out,
-                     model=['Website'], batchsize=3)
```
Comment (PLNech) [inline tests/test_commands.py]: why removing batchsize? :thinking:

File: tests/models.py [hunk 2]
```diff
@@ -70,5 +70,8 @@ def property_string(self):
 
 
 class BlogPost(models.Model):
-    author = models.ForeignKey(User)
+    author = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE
```
Comment (PLNech) [inline tests/models.py]: What required this change?

File: tests/test_commands.py (hunk 1 above)
Comment (clemfromspace) [reply tests/test_commands.py]: The `clearindex` command do not take a `batchsize` argument ;)
(The extra arguments are now checked in Django > 2.0, so the test fail on this version, for a good reason)

File: tests/models.py (hunk 2 above)
Comment (clemfromspace) [reply tests/models.py]: In Django>2, the `on_delete` argument on a `ForeignKey` is now required: https://docs.djangoproject.com/en/2.0/releases/2.0/#features-removed-in-2-0

**Pull request merged**